cilium/cmd/debuginfo.go

```diff
@@ -77,14 +96,82 @@ var sections = map[string]addSection{
 
 func init() {
 	rootCmd.AddCommand(debuginfoCmd)
-	debuginfoCmd.Flags().StringVarP(&file, "file", "f", "", "Redirect output to file")
-	debuginfoCmd.Flags().StringVarP(&html, "html-file", "", "", "Convert default output to HTML file")
+	debuginfoCmd.Flags().BoolVarP(&outputToFile, "file", "f", false, "Redirect output to file(s)")
 	debuginfoCmd.Flags().BoolVarP(&filePerCommand, "file-per-command", "", false, "Generate a single file per command")
+	debuginfoCmd.Flags().StringSliceVar(&outputOpts, "output", []string{}, "markdown| html| json| jsonpath='{}'")
+	debuginfoCmd.Flags().StringVar(&outputDir, "output-directory", "", "directory for files (if specified will use directory in which this command was ran)")
+}
+
+func validateInput() {
+	if outputDir != "" && !outputToFile {
+		fmt.Fprintf(os.Stderr, "invalid option combination; specified output-directory %q, but did not specify for output to be redirected to file; exiting\n", outputDir)
+		os.Exit(1)
+	}
+	validateOutputOpts()
+}
+
+func validateOutputOpts() {
+
+	for _, outputOpt := range outputOpts {
+		switch strings.ToUpper(outputOpt) {
```

Comment:
Somewhat arbitrary, but as a matter of consistency we tend to use `ToLower()` far more often than `ToUpper()`:

```
$ demo
$ git grep ToUpper -- :"^vendor" :"^api" | wc -l
3
$ git grep ToLower -- :"^vendor" :"^api" | wc -l
31
```